tool/tsh/common/vnet_darwin.go

```diff
@@ -80,6 +80,12 @@ type vnetAdminSetupCommand struct {
 	ipv6Prefix string
 	// dnsAddr is the IP address for the VNet DNS server.
 	dnsAddr string
+	// egid of the user starting VNet. Unsafe for production use, as the egid comes from an unstrusted
+	// source.
+	egid int
+	// euid of the user starting VNet. Unsafe for production use, as the euid comes from an unstrusted
+	// source.
+	euid int
```

Comment:
Unlike, say, `sudo`, `osascript -e "do shell script \"…\" with administrator privileges"` does not seem to offer `SUDO_UID` or any other way to identify the user who spawned osascript in the first place.

`osascript -e "do shell script \"id -ru\" with administrator privileges"` returns `0`.

However, after releasing the launch daemon we don't plan to use osascript in production (since we compile tsh with `VNETDAEMON=yes`). It's there only to make development easier. As such, I pass egid and euid through argv.